### PR TITLE
fix(RHINENG-9813) Enable searching for tags that contain "$"

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [RHINENG-9813](https://issues.redhat.com/browse/RHINENG-9813).
Adds `$` to the list of regex-escaped characters. Also adds a test for this scenario.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Include raw query examples in the PR description, if adding/modifying SQL query
- [x] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: api/__init__.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 STATUS_CODE = "status_code"
 PROCESSING_TIME = "processing_time"
 
-ESCAPE_CHARS = {c for c in '.?+*|{}[]()"\\#@&<>~'}
+ESCAPE_CHARS = {c for c in '.?+*|{}[]()"\\#@&<>~$'}
```
Comment (vkrizan) [inline api/__init__.py:18]: I wonder why this is a set and not just a string. Those are iterable by character. Hashing is slower.

**Review by jpramos123 — DISMISSED** [review]

**Review by roliveri — DISMISSED** [review]
LGTM

File: api/__init__.py (hunk 1 above)
Comment (kruai) [reply api/__init__.py]: IDK, good to know. Changed it to a string.

**Review by thearifismail — COMMENTED** [review]
LGTM

**Review by thearifismail — COMMENTED** [review]
LGTM

**Pull request merged**